### PR TITLE
Add transformation aggregation replacement logic

### PR DESCRIPTION
**Pull request by lkuchenb**
This sub-PR adds the migration of the current aggregate transformation logic.

### CODE REVIEW

**Review by KerstenBreuer — COMMENTED** [review]
I am sorry, but I cannot give good feedback because I struggle to understand the proposed transformations. I would be happy to go through them in a meeting. Otherwise I also trust you that they are what we need conceptually. However, we might still want to agree on common naming patterns. An whether transformation can be used to perform a batch of tasks (e.g. modify multiple classes) or shall only focus at one task at a time.

File: 50-dhole/technical_specification.md [hunk 1]
```diff
@@ -240,10 +240,134 @@ B. For merging relation properties:
...
+- class: MyClass
+  target_path: [path, of, properties]
+  schema:
+    type: "object"
+    additionalProperties: false
+  value: {}
+
+# Examples from current config
+- class: Dataset
+  target_path: [datasets]
```
[the remark this answers is not in the document]
Comment (lkuchenb) [reply 50-dhole/technical_specification.md]: Fixed, that was wrong indeed.

File: 50-dhole/technical_specification.md [hunk 2]
```diff
@@ -240,10 +240,134 @@ B. For merging relation properties:
...
+#### Transformation 1: Add Subschema
+
+A transformation that enables the insertion of a subschema into a new property
+of any object within the current content schema of a class, including an initial
+value. The configuration may look as follows:
+
+```yaml
+# Generic Example
+- class: MyClass
+  target_path: [path, of, properties]
```
[the remark this answers is not in the document]
Comment (lkuchenb) [reply 50-dhole/technical_specification.md:257]: Yes, data could look like that after the insertion of the subschema.

File: 50-dhole/technical_specification.md [hunk 3]
```diff
@@ -240,10 +240,134 @@ B. For merging relation properties:
...
+```yaml
+- class: Dataset
+  target_path: [samples_summary, count]
+  source_path: [samples]
+```
+
+####  Transformation 3: Count content values
+
+* The transformation shall count the values encountered at a specified property in the content of an object.
+* The transformation shall validate that at least one of the traversed references is multi-valued by its cardinality or one of the traversed content elements is an array.
```
[the remark this answers is not in the document]
Comment (lkuchenb) [reply 50-dhole/technical_specification.md:302]: Sorry for the wrong wording, references are relations, I will update the spec accordingly. In order to be able to count anything, the counting transformations shall validate that there is actually a case of multiplicity anywhere in the source path, either during relation traversal or during content traversal.

**Comment by lkuchenb** [comment]
@KerstenBreuer Ready for another round I think

File: 50-dhole/technical_specification.md [hunk 4]
```diff
@@ -240,10 +240,141 @@ B. For merging relation properties:
...
+##### Configuration
+
+```yaml
+# Generic Example, defaults shown for optional parameters
+- class_name: MyClass
+  target_content:
+    object_path: "" # Optional, default empty string refers to content root object
+    property_name: new_property
+    required: true # Optional
+  schema: # Optional
```
Comment (KerstenBreuer) [inline 50-dhole/technical_specification.md:263]: Right?
```suggestion
  schema: # Optional, defaults to the following value
```

File: 50-dhole/technical_specification.md [hunk 5]
```diff
@@ -240,10 +240,141 @@ B. For merging relation properties:
...
+
+A transformation that enables the insertion of a subschema into a new property
+of any object within the current content schema of a class, including an initial
+value. The schema will be modified by inserting the subschema at the specified path. The data will be modified by inserting the default value at the specified path. The user may control whether or not the newly added property is added to the parent property's `required` list.
+
+##### Configuration
+
+```yaml
+# Generic Example, defaults shown for optional parameters
+- class_name: MyClass
```
Comment (KerstenBreuer) [inline 50-dhole/technical_specification.md:258]: Should we stay with arrays of instructions for our transformation steps? I am also happy to defer that decision to later.

File: 50-dhole/technical_specification.md [hunk 6]
```diff
@@ -240,10 +240,141 @@ B. For merging relation properties:
...
+    property_name: datasets
+
+- class_name: Dataset
+  target_content:
+    property_name: "studies_summary"
+```
+
+#### Transformation 2: Count References
+
+* The transformation shall count how many target objects are referenced from each source object given the reference name.
```
Comment (KerstenBreuer) [inline 50-dhole/technical_specification.md:280]:
```suggestion
* The transformation shall count how many target objects are referenced from each source object given the relation name.
```

File: 50-dhole/technical_specification.md [hunk 7]
```diff
@@ -240,10 +240,141 @@ B. For merging relation properties:
...
+* The transformation shall validate whether the target is defined with multiplicity and fail otherwise
+
+Example config:
+
+```yaml
+- class_name: Dataset
+  target_content:
+    object_path: samples_summary
+    property_name: count
+  source_relation_path: "Dataset(samples)>Sample"
```
Comment (KerstenBreuer) [inline 50-dhole/technical_specification.md:290]: Great, actually I just thought that in deed passive relations (containing <) might also be interesting to count. This notation would enable it.

**Review by KerstenBreuer — APPROVED** [review]
Looks good, looking forward ot see it in action.

File: 50-dhole/technical_specification.md (hunk 5 above)
Comment (KerstenBreuer) [reply 50-dhole/technical_specification.md:258]: As discussed, will keep the current representation.

**Pull request merged**